src/Export/Adapters/PropertiesAdapter.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\FinSearch\Export\Adapters;
+
+use FINDOLOGIC\Export\Data\Property;
+use FINDOLOGIC\FinSearch\Utils\Utils;
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Framework\DataAbstractionLayer\Pricing\Price as ProductPrice;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class PropertiesAdapter
+{
+    /** @var array  */
+    private $properties = [];
```

Comment:
Adapters should not be state aware and this is dangerous. If you would call this adapter multiple times, you will pollute it more and more, which not only only gives you incorrect data, but may eventually also cause a memory-leak. Please don't store any data in a service.